app/routes.test.js

```diff
@@ -300,6 +303,89 @@ describe('routes', () => {
       }));
   });
 
+  describe('POST /organisation/neworder/call-off-ordering-party', () => {
+    const path = '/organisation/order-id/call-off-ordering-party';
+
+    afterEach(() => {
+      orderingPartyController.putCallOffOrderingParty.mockReset();
+    });
+
+    it('should return 403 forbidden if no csrf token is available', () => (
+      testPostPathWithoutCsrf({
+        app: request(setUpFakeApp()), pathToTest: path, mockAuthorisedCookie,
+      })
+    ));
+
+    it('should redirect to the login page if the user is not logged in', () => (
+      testAuthorisedPostPathForUnauthenticatedUser({
+        app: request(setUpFakeApp()),
+        csrfPagePath: path,
+        pathToTest: path,
+        mockAuthorisedCookie,
+        expectedRedirectPath: 'http://identity-server/login',
+      })
+    ));
+
+    it('should show the error page indicating the user is not authorised if the user is logged in but not authorised', () => (
+      testAuthorisedPostPathForUnauthorisedUsers({
+        app: request(setUpFakeApp()),
+        csrfPagePath: path,
+        pathToTest: path,
+        mockAuthorisedCookie,
+        mockUnauthorisedCookie,
+        expectedPageId: 'data-test-id="error-title"',
+        expectedPageMessage: 'You are not authorised to view this page',
+      })
+    ));
+
+    it('should return the correct status and text if response.success is true', async () => {
+      orderingPartyController.putCallOffOrderingParty = jest.fn()
+        .mockImplementation(() => Promise.resolve({ success: true }));
+
+      const { cookies, csrfToken } = await getCsrfTokenFromGet({
+        app: request(setUpFakeApp()), csrfPagePath: path, mockAuthorisedCookie,
+      });
+
+      return request(setUpFakeApp())
+        .post(path)
+        .type('form')
+        .set('Cookie', [cookies, mockAuthorisedCookie])
+        .send({
+          description: 'a description of the order',
```

Comment:
don't think these affects the test but suppose this should be call-off ordering party data instead. Or just remove it all together so it doesn't get confusing

Reply:
ah yeah